daemon/api.go

```diff
@@ -55,7 +55,7 @@ import (
 	"github.com/snapcore/snapd/store"
 )
 
-var api = []*Command{
+var privateAPI = []*Command{
```

Comment:
The separation here doesn't feel ideal. Can we find a way to do something like this:
```
        snapctlCmd = &Command{
                Path:   "/v2/snapctl",
                SnapOK: true,
                POST:   runSnapctl,
        }
```

and share everything else?  We should be able to keep all commands together in the same API handling, and use exactly the same router and all. It's just whether we allow requests through a particular socket to go in or not. This also means we can share endpoints across the two APIs
very easily.